source/GameInterface/Services/Villages/Patches/ChangeVillageStatePatch.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Common;
+using Common.Extensions;
+using Common.Messaging;
+using Common.Util;
+using GameInterface.Services.Villages.Messages;
+using HarmonyLib;
+using System;
+using System.Reflection;
+using TaleWorlds.CampaignSystem.Actions;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.CampaignSystem.Settlements;
+
+namespace GameInterface.Services.Villages.Patches
+{
+    [HarmonyPatch(typeof(ChangeVillageStateAction), "ApplyInternal")]
```

Comment:
Instead sync `VillageState` setter

You can keep this patch but update prefix to `private static bool Prefix() => ModInformation.IsServer`